web-service-consumer/1.6/modules/ROOT/pages/web-service-consumer-troubleshooting.adoc

```diff
@@ -0,0 +1,166 @@
+
+== Enable Wire Logging
+
+When troubleshooting Web Service Consumer Connector it is helpful to look at the transport request and response on the Web Service Consumer.
+If you are using the underlying HTTP transport, you can enable wire logging to see the exact HTTP message, the headers used in the transmission and the status of the response.
+
+To enable wire logging:
+
+. Access Anypoint Studio, navigate to the *Package Explorer* view.
+. Open your application's project name.
+. Open the *src/main/resources* path folder.
+. Open the `log4j2.xml` file inside the folder.
+. Add or uncomment the following line:
++
+[source,xml,linenums]
+----
+<AsyncLogger name="org.mule.service.http.impl.service.HttpMessageLogger" level="DEBUG" />
+----
++
+. Save all your application changes.
+.  Click the project name in *Package Explorer* and then click *Run* > *Run As* > *Mule Application*.
+. Navigate to the *Console* view to read the HTTP messages in the Mule's logger, for example:
++
+
+[source,plain-text]
+----
+POST /essentials/delta HTTP/1.1
+soapaction:
+Host: ilt.mulesoft-training.com
+User-Agent: AHC/1.0
+Connection: keep-alive
+Accept: */*
+Content-Type: text/xml; charset=UTF-8
+Transfer-Encoding: chunked
+
+b1
+<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/"><soap:Body><ns0:listAllFlights xmlns:ns0="http://soap.training.mulesoft.com/"/></soap:Body></soap:Envelope>
+
+DEBUG 2021-04-06 15:04:23,058 [[wsc-base].wsc-dispatcher.11 SelectorRunner] [processor: ; event: ] org.mule.service.http.impl.service.HttpMessageLogger.wsc-dispatcher: REQUESTER
+0
+
+
+DEBUG 2021-04-06 15:04:23,319 [[wsc-base].wsc-dispatcher.11 SelectorRunner] [processor: ; event: ] org.mule.service.http.impl.service.HttpMessageLogger.wsc-dispatcher: REQUESTER
+HTTP/1.1 200
+Date: Tue, 06 Apr 2021 18:04:23 GMT
+Content-Type: text/xml; charset=UTF-8
+Transfer-Encoding: chunked
+Connection: keep-alive
+MULE_ENCODING: UTF-8
+Strict-Transport-Security: max-age=31536000; includeSubdomains;
+X-Content-Type-Options: 1
+X-Frame-Options: SAMEORIGIN
+X-XSS-Protection: 1; mode=block
+
+bfe
+<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/"><soap:Body><ns2:listAllFlightsResponse xmlns:ns2="http://soap.training.mulesoft.com/">...
+----
+
+== Troubleshoot Access Attachments
+
+Some servers can respond with MTOM messages. These messages separate their data into parts which enables different content types of data to coexist in the same HTTP message.
+
+MTOM is in charge of sending binary data in a separate part and referencing the part from the corresponding field within the SOAP envelope.
+Web Service Consumer users interact with these messages (multipart) only with the MTOM configuration. This is not to say that MTOM, XOP, and MIME or multipart are the same.
```

Comment:
Can you provide some transitions in lines 64-67? The text goes from introducing MTOM as able to combine diff. types of info by using parts. Then it talks about a separate binary data part, and SOAP. Then it returns to the multipart message and Web Service (so why the separate, binary part). What does "only with the MTOM configuration" mean: what part of all this uses the MTOM configuration, and what do we mean by that configuration (binary part, multiple parts)? Finally, why go off topic to mention XOP, MIME, and "multipart" (and why is "multipart" separate from MTOM)?

Reply:
If I had to guess, I'd say that you took this info from the sources without reworking it your Kevo way!